### PR TITLE
Collinear quark Changes

### PR DESCRIPTION
**Pull request by vibhutittu**
Modified the cleanCollinVars method.

### CODE REVIEW

File: source/quark/R/01_QuarkData.R [hunk 1]
```diff
@@ -472,11 +472,62 @@ QuarkData$methods(
...
+		} else if(nrow(maxNaCount)==1){
+        
+        maxNaCount<- as.character(maxNaCount$var1)
+        varCount<- varCount[-which(varCount$Var1 == maxNaCount),]
+        collinVarPairs<- subset(collinVarPairs,collinVarPairs[,1]!=maxNaCount)
+        collinVarPairs<- subset(collinVarPairs,collinVarPairs[,2]!=maxNaCount)
+        vector_a<- append(vector_a,maxNaCount)
+      }
+      
+    }else
```
Comment (kylelang) [inline source/quark/R/01_QuarkData.R]: Put 'else if' on a single line.

File: source/quark/R/01_QuarkData.R [hunk 2]
```diff
@@ -472,11 +472,62 @@ QuarkData$methods(
...
 
-    cleanCollinVars = function(x)                                               {
-        "Remove one variable from all collinear pairs"
-        collinVars <<- x
-        removeVars(x = unique(collinVars$var1), reason = "collinear")
-    },
+    
+  cleanCollinVars = function(x)                                                {
+       "Remove one variable from all collinear pairs"
+        collinVars <- x
```
Comment (kylelang) [inline source/quark/R/01_QuarkData.R]: You must use the deep assignment operator '<<-' to assign values to the fields of a reference class object. The normal assignment operator '<-' will only make a local assignment when employed within a reference class method (i.e., the underlying field is not modified). 

Please change line 478 to 'collinVars <<- x'

File: source/quark/R/01_QuarkData.R [hunk 3]
```diff
@@ -472,11 +472,62 @@ QuarkData$methods(
...
+        collinVarPairs <- collinVars[,1:2]
+  
+  while(nrow(collinVarPairs)>0){
+    
+    varCount<- data.frame(table(unlist(collinVarPairs)))
+    maxVarCount<- varCount[which(varCount$Freq==max(varCount$Freq)),]
+    
+    naCount <-sapply(data,function(c) sum(length(which(is.na(c)))))
```
Comment (kylelang) [inline source/quark/R/01_QuarkData.R]: You don't need to count NAs. Use the complement of the 'respCounts' field.

**Review by kylelang — CHANGES_REQUESTED** [review]
In addition to the specific changes requested in the embedded comments, please format your code to conform to the official style guide, and please remove the 'Experiment' directory.

File: source/quark/R/01_QuarkData.R [hunk 4]
```diff
@@ -78,7 +78,10 @@ QuarkData <- setRefClass("QuarkData",
                              nImps        = "integer",
                              compFormat   = "character",
                              miDatasets   = "ANY",
-                             miceObject   = "ANY"
+                             miceObject   = "ANY",
+			     sameNaCntVec = "vector",
```
Comment (kylelang) [inline source/quark/R/01_QuarkData.R]: Do these three objects need to be fields in the QuarkData object? It looks like they are only used locally in the cleanCollinVars() function. Do they need to persist after cleanCollinVars() returns? If not please remove these fields and treat these variables as local.

File: source/quark/R/01_QuarkData.R [hunk 5]
```diff
@@ -472,11 +481,71 @@ QuarkData$methods(
...
+    
+        varCount     <- data.frame(table(unlist(collinVarPairs)))
+	maxVarCount  <- varCount[which(varCount$Freq == max(varCount$Freq)), ]
+	maxVarCommon <- intersect(names(respCounts), 
+		                          maxVarCount[, 1])
+    
+        ## Check for missing value counts if maximum counts are equal
+        if(nrow(maxVarCount) > 1){
+      
+        maxNaCountValue  <- max(respCounts[maxVarCommon])
```
Comment (kylelang) [inline source/quark/R/01_QuarkData.R]: The 'respCounts' field is a vector of variablewise response rates (i.e., the complement of the variablewise missing data rates). This line will find the variable for which maxVarCommon == TRUE with the least missing data. Is that what you intend?

File: source/quark/R/01_QuarkData.R [hunk 6]
```diff
@@ -472,11 +481,71 @@ QuarkData$methods(
...
+	collinVarPairs  <- subset(collinVarPairs, 
+	                             collinVarPairs[,1] != firstVar)
+        collinVarPairs  <- subset(collinVarPairs, 
+	                             collinVarPairs[,2] != firstVar)
+								 
+        diffMaxCntVec   <- append(diffMaxCntVec, firstVar)
+                  }
+            }
+			
+        vector     <- c(sameNaCntVec,diffNaCntVec, diffMaxCntVec)
```
Comment (kylelang) [inline source/quark/R/01_QuarkData.R]: 'vector' is the name of a built-in R function (i.e., the one that creates vectors). We don't want to use extant function names as variable names. Please rename this variable.

**Review by kylelang — CHANGES_REQUESTED** [review]

File: source/quark/R/01_QuarkData.R (hunk 4 above)
Comment (vibhutittu) [reply source/quark/R/01_QuarkData.R]: Yes these three variables are used locally inside cleanCollinVars(). They will not persist after cleanCollinVars() but they need to be initialised locally . So shall I initilaize them locally inside the method cleanCollinVars()

File: source/quark/R/01_QuarkData.R (hunk 5 above)
Comment (vibhutittu) [reply source/quark/R/01_QuarkData.R]: I am finding the variables in "maxVarCommon" that exist in both respCounts and maxVarCounts and then the maximum missing count value out of those and store in maxNaCountValue.

File: source/quark/R/01_QuarkData.R (hunk 6 above)
Comment (vibhutittu) [reply source/quark/R/01_QuarkData.R]: Ok I will rename it.

**Review by kylelang — APPROVED** [review]

**Pull request merged**